46.md

```diff
@@ -2,161 +2,67 @@ NIP-46
 ======
 
 Nostr Connect
-------------------------
+-------------
 
-`draft` `optional` `author:tiero` `author:giowe` `author:vforvalerio87`
+`draft` `optional` `author:tiero` `author:giowe` `author:vforvalerio87` `author:pablof7z` `author:fiatjaf`
 
-## Rationale
+This NIP describes a method for 2-way communication between a **remote signer** and a normal Nostr client. The remote signer could be, for example, a hardware device dedicated to signing Nostr events, while the client is a normal Nostr client.
 
-Private keys should be exposed to as few systems - apps, operating systems, devices - as possible as each system adds to the attack surface.
+The remote signer generates a connection token in the form
 
-Entering private keys can also be annoying and requires exposing them to even more systems such as the operating system's clipboard that might be monitored by malicious apps.
-
-
-## Terms
-
-* **App**: Nostr app on any platform that *requires* to act on behalf of a nostr account.
-* **Signer**: Nostr app that holds the private key of a nostr account and *can sign* on its behalf.
-
-
-## `TL;DR`
-
-
-**App** and **Signer** sends ephemeral encrypted messages to each other using kind `24133`, using a relay of choice. 
-
-App prompts the Signer to do things such as fetching the public key or signing events.
+```
+<npub1...>#<secret>?relay=wss://...&relay=wss://...
+```
 
-The `content` field must be an encrypted JSONRPC-ish **request** or **response**.
+The user copies that token and pastes it in the client UI somehow. Then the client can send events of kind `24133` to the specified relays and wait for responses from the remote signer.
 
-## Signer Protocol
+The `secret` allows the signer to identify the client's key and, perhaps, assign it certain permissions (e.g. *sign only event `kind:7`, reject encrypt/decrypt requests*)
 
-### Messages
+Event payloads are [NIP-04](04.md)-encrypted JSON blobs that look like JSONRPC.
 
-#### Request
+Events sent by the client to the remote signer have the following format:
 
-```json
+```js
 {
-  "id": <random_string>,
-  "method": <one_of_the_methods>,
-  "params": [<anything>, <else>]
+  "pubkey": "<client-deterministically-generated-key-for-the-connection>"
```

Comment:
wait is it really just for one connection? shouldn't we allow reuse of the client's keys for multiple connections?

Reply:
Yes, I believe we allow anything.